tests/e2e/README.md

```diff
@@ -0,0 +1,135 @@
+
+#### Testing the operator available for the VERSION (default)
+
+In this scenario, you want to run the E2E test suite against the operator image corresponding to the `VERSION` declared in the project [`Makefile`](../../Makefile), which should be publicly available at `quay.io/janus-idp/operator:<VERSION>`.
+
+This is the default behavior.
+
+This should work on any Kubernetes or OpenShift cluster:
+
+```shell
+$ make test-e2e
+```
+
+#### Testing a specific image (e.g. PR image)
+
+In this scenario, you want to run the E2E test suite against an existing operator image.
+
+This should work on any Kubernetes or OpenShift cluster:
+
+```shell
+# if the tag is already published and available at the default location: quay.io/janus-idp/operator
+$ make test-e2e VERSION=0.2.0-3d1c1e0
+
+# or you can override the full image repo name
+$ make test-e2e IMG=my.registry.example.com/operator:0.2.0-3d1c1e0
+```
+
+Note that `VERSION` and `IMG` override the respective variables declared in the project [`Makefile`](../../Makefile).
+
+#### Building and testing local changes on supported local clusters
+
+In this scenario, you are iterating locally, and want to run the E2E test suite against your local changes. You are already using a local cluster like [`kind`](https://kind.sigs.k8s.io/), [`k3d`](https://k3d.io/) or [`minikube`](https://minikube.sigs.k8s.io/docs/), which provide the ability to import images into the cluster nodes.
+
+To do so, you can:
+1. set `BACKSTAGE_OPERATOR_TESTS_BUILD_IMAGES` to `true`, which will result in building the operator image from the local changes,
+2. and set `BACKSTAGE_OPERATOR_TESTS_PLATFORM` to a supported local cluster, which will result in loading the image built directly in that cluster (without having to push to a separate registry).
+
+##### `kind`
+
+```shell
+$ kind create cluster
+$ make test-e2e \
+    BACKSTAGE_OPERATOR_TESTS_BUILD_IMAGES=true \
+    BACKSTAGE_OPERATOR_TESTS_PLATFORM=kind
+```
+
+##### `k3d`
+
+```shell
+$ k3d cluster create
+$ make test-e2e \
+    BACKSTAGE_OPERATOR_TESTS_BUILD_IMAGES=true \
+    BACKSTAGE_OPERATOR_TESTS_PLATFORM=k3d
+```
+
+##### `minikube`
+
+```shell
+$ minikube start
+$ make test-e2e \
+    BACKSTAGE_OPERATOR_TESTS_BUILD_IMAGES=true \
+    BACKSTAGE_OPERATOR_TESTS_PLATFORM=minikube
+```
+
+#### Testing a specific version using OLM
+
+In this scenario, you want to leverage the [Operator Lifecycle Manager (OLM)](https://olm.operatorframework.io/) to deploy the Operator.
+
+This requires OLM to be installed in the cluster.
+
+```shell
+$ make test-e2e BACKSTAGE_OPERATOR_TEST_MODE=olm
+```
+
+#### Testing a downstream build of Red Hat Developer Hub (RHDH)
+
+In this scenario, you want to run the E2E tests against a downstream build of RHDH.
+
+This works only against OpenShift clusters. If testing a CI build, please follow the instructions in [Installing CI builds of Red Hat Developer Hub](../../.rhdh/docs/installing-ci-builds.adoc) to add your Quay token to the cluster.
```

Comment:
If it needs that this Openshift cluster be configured as a current context of your kubeconfig, could you please describe this a bit... It is not really clear how 'make test-e2e BACKSTAGE_OPERATOR_TEST_MODE=rhdh-latest' knows where this Openshift is :)

Reply:
But this is out of scope of this document, no? Anyway, I've added a note and link to help clarify this. See [`ff4daa1` (#204)](https://github.com/janus-idp/operator/pull/204/commits/ff4daa155c568585ea6615bcf15ac89d4eac797e)

Reply:
I am just trying to make it clear for myself... not sure about the doc scoping :) 
I think description counle be as smple as: "Make sure your current context points to.....   calling 'kubectl config current-context'"

Reply:
Okay. See [`2b5b821` (#204)](https://github.com/janus-idp/operator/pull/204/commits/2b5b82184c8115b02d565a9ca4a42b54bc2a0cd2)

Reply:
If you are talking about "To run the end-to-end tests, make sure you have an active connection to a cluster in your current [kubeconfig]"
I am just not sure about terminology like "active connection" in this context :) , to me "current context " sounds more familiar in this context.
But if you think it is clear enough - go ahead, I am not blocking it.

Reply:
> If you are talking about "To run the end-to-end tests, make sure you have an active connection to a cluster in your current [kubeconfig]"
I am just not sure about terminology like "active connection" in this context :) , to me "current context " sounds more familiar in this context.

Yeah, I borrowed the sentence "active connection" from the OpenShift doc. Then added a note on how to check the current context, as you suggested ;-)

> But if you think it is clear enough - go ahead, I am not blocking it.

Okay, please approve the PR then if it is good for you ^^. Thanks.